dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/RequestItemRepository.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+
+package org.dspace.app.rest.repository;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.dspace.app.requestitem.RequestItem;
+import org.dspace.app.requestitem.service.RequestItemService;
+import org.dspace.app.rest.converter.RequestItemConverter;
+import org.dspace.app.rest.exception.RepositoryMethodNotImplementedException;
+import org.dspace.app.rest.exception.UnprocessableEntityException;
+import org.dspace.app.rest.model.RequestItemRest;
+import org.dspace.app.rest.projection.DefaultProjection;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Item;
+import org.dspace.content.service.BitstreamService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+
+/**
+ * Component to expose item requests.
+ *
+ * @author Mark H. Wood <mwood@iupui.edu>
+ */
+@Component(RequestItemRest.CATEGORY + '.' + RequestItemRest.NAME)
+public class RequestItemRepository
+        extends DSpaceRestRepository<RequestItemRest, String> {
+    @Autowired(required = true)
+    protected RequestItemService requestItemService;
+
+    @Autowired(required = true)
+    protected BitstreamService bitstreamService;
+
+    @Autowired(required = true)
+    protected ItemService itemService;
+
+    @Autowired(required = true)
+    protected RequestItemConverter requestItemConverter;
+
+    @Override
+    //@PreAuthorize(expr)
+    public RequestItemRest findOne(Context context, String id) {
```

Comment:
@mwoodiupui : Regarding your previous comment about security, additional security could/should be added here.  This likely should have a `@PreAuthorize("hasAuthority('AUTHENTICATED')")` -- to verify the user is authenticated.   We also should have a test (in the IT) which verifies that an unauthenticated user cannot access any individual requests.

(Sidenote: in your comment you said the token is an `int`, but here it's shown to be a String.  It appears to me it's actually a random hex key, see https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/app/requestitem/RequestItemServiceImpl.java#L45)

Reply:
Thanks for the suggested expression.

v6 behavior seems to have been to send the user to the login page if not logged in.

I don't know where I got the `int`, unless it is the debug log message in `RequestItemServiceImpl#createRequest` which says it created a "requestitem_token" followed by the RequestItem's ID (which is Integer).  (I'm going to fix that.)  Anyway, random hex is at least a good deal harder to guess.

Reply:
@tdonohue @benbosman I've been re-thinking the permission issue.  https://wiki.lyrasis.org/display/DSDOC7x/Request+a+Copy describes the feature.  The whole point of Request a Copy is to cause DSpace to punt the access control decision to a human.  The request form collects contact information, so we don't need an EPerson to supply it.  Request a Copy operates outside of DSpace's normal access control -- the request is routed to someone determined by the DSpace configuration, and granting it causes the file(s) to be sent.

All that to say this:  probably this feature should *not* interact with DSpace access control at all.

I must've been tired when I wrote my previous post.  A new look at [the XMLUI code ](https://github.com/DSpace/DSpace/blob/da3154b04511389b3886207ed91d0d18a0adaaa4/dspace-xmlui/src/main/java/org/dspace/app/xmlui/aspect/artifactbrowser/ItemRequestForm.java#L150) seems to show that, if the session is not logged in, the request form will provide a link to login *if desired*, but the form is presented; however, if the session *is* logged in *and* has READ access to the Item, the browser is redirected to the item page.  Unauthenticated users, and authenticated users without READ, get the form and may use it to send a request to bypass DSpace access control.

Reply:
@mwoodiupui : In the DSpace 7 realm, I'm a little worried about providing an unauthenticated REST endpoint where anyone/anything can kickoff a ton of "requests for copies" anonymously (that seems like an invitation for a DOS attack).  That's why I wonder if it's more appropriate to place this behind a requirement to authenticate, even if it slightly changes how it works.  

So, for instance:
1. Request a Copy can only be submitted by authenticated users.  They can perhaps change their email address if they want the response sent back to a different email address. 
2. Request a Copy can be approved by anyone with the generated (random) link (no need to login to approve).

That would in turn mean that the `findOne()` method needs to allow for anonymous access (for a user to be able to approve it anonymously).  So, that means it should have a `@PreAuthorize("permitAll()")`.

However, the `createAndReturn()` method would need to require authentication...i.e. `@PreAuthorize("hasAuthority('AUTHENTICATED')")`... to force you to be authenticated before you can create a new "request for copy".

Reply:
That change is not "slight", IMHO.  As I recall, requiring the requester to have an account was specifically rejected during the initial discussion of alternatives to this feature.  What is the typical use case?  (Ours is precisely the case of anonymous users to be vetted by those controlling access.)  I think such changes need wider discussion.

We might be better off with something like a rate limit.

Reply:
@mwoodiupui : That's reasonable then.  To be honest, now that I think about this again, since it's a POST request, I'm realizing this `createAndReturn` would require CORS & CSRF to pass....so, the requests must already be coming from a "trusted" source/origin, which also must be passing a valid CSRF token.  This provides us with a bit of a safety net already that it's not just anyone from anywhere generating these "requests for copies".

So, I'm OK with your approach. I'd just ask that you explicitly add a `@PreAuthorize("permitAll()")` to both methods and add ITs which prove the behavior you expect.  If we run into later issues with this behavior we can fix it at a later time.